fw/ws2812.c

```diff
@@ -0,0 +1,194 @@
+/*
+* light weight WS2812 lib V2.0b
+*
+* Controls WS2811/WS2812/WS2812B RGB-LEDs
+* Author: Tim (cpldcpu@gmail.com)
+*
+* Jan 18th, 2014  v2.0b Initial Version
+* Nov 29th, 2015  v2.3  Added SK6812RGBW support
+*
+* License: GNU GPL v2+ (see License.txt)
+*/
+
+#include "ws2812.h"
+#include <avr/interrupt.h>
+#include <avr/io.h>
+#include <util/delay.h>
+
+// Normally ws2812_sendarray_mask() runs under disabled-interrupt condition,
+// undefine if you want to accept interrupts in that function.
+#define interrupt_is_disabled
+
+// Setleds for standard RGB
+void inline ws2812_setleds(struct cRGB *ledarray, uint16_t leds)
+{
+   ws2812_setleds_pin(ledarray,leds, _BV(ws2812_pin));
+}
+
+void inline ws2812_setleds_pin(struct cRGB *ledarray, uint16_t leds, uint8_t pinmask)
+{
+  ws2812_sendarray_mask((uint8_t*)ledarray,leds+leds+leds,pinmask);
+  _delay_us(ws2812_resettime);
```

Comment:
do you think this would cause any issues with the asynchronous nature of this program?

Reply:
no idea
FWIW `ws2812_resettime` is set to be 50 microseconds (so 0.05 milliseconds I believe)

Reply:
ok, something to keep in mind for the future that there is some usage of `_delay_us` pulled in by this lib